src/components/CommentItem/CommentItem.tsx

```diff
@@ -0,0 +1,49 @@
+import { useContext } from 'react';
+import { removeComments } from '../../api/comments';
+import { Comment } from '../../types/Comment';
+import { MainContext } from '../MainContext/MainContext';
+
+type Props = {
+  comment: Comment,
+};
+
+export const CommentItem: React.FC<Props> = ({ comment }) => {
+  const { comments, setComments } = useContext(MainContext);
+
+  const {
+    id, email, name, body,
+  } = comment;
+
+  const handleRemoveComment = (commentId: number) => {
+    setComments(comments?.filter(elem => elem.id !== commentId));
```

Comment:
work with previous value
```suggestion
    setComments(prevComments => prevComments?.filter(elem => elem.id !== commentId));
```